dds/DCPS/Message_Block_Ptr.h

```diff
@@ -22,6 +22,42 @@ struct Message_Block_Deleter
 
 typedef unique_ptr<ACE_Message_Block,Message_Block_Deleter> Message_Block_Ptr;
 
+class OpenDDS_Dcps_Export Message_Block_Shared_Ptr {
+public:
+  Message_Block_Shared_Ptr() {}
+
+  explicit Message_Block_Shared_Ptr(ACE_Message_Block* payload)
+    : msg_(payload)
+  {}
+
+  Message_Block_Shared_Ptr(const Message_Block_Shared_Ptr& other)
+    : msg_(ACE_Message_Block::duplicate(other.msg_.get()))
+  {}
+
+  Message_Block_Shared_Ptr& operator=(const Message_Block_Shared_Ptr& other)
+  {
+    Message_Block_Shared_Ptr cpy(other);
+    swap(*this, cpy);
+    return *this;
+  }
+
+  explicit operator bool() const { return msg_.get(); }
+
+  ACE_Message_Block& operator*() const { return *msg_; }
+
+  ACE_Message_Block* operator->() const { return msg_.get(); }
+
+  ACE_Message_Block* get() const { return msg_.get(); }
+
+  static void swap(Message_Block_Shared_Ptr& a, Message_Block_Shared_Ptr& b)
+  {
+    std::swap(a.msg_, b.msg_);
```

Comment:
This needs be an unqualified swap for ADL.  Looks to be the cause of some of the CI build problems.  On some platforms without unique_ptr, it is emulated in the OpenDDS header.
https://en.cppreference.com/w/cpp/named_req/Swappable